satellite/marketing/marketingweb/server.go

```diff
@@ -28,16 +41,33 @@ type Config struct {
 // Server represents marketing offersweb server
 type Server struct {
 	log *zap.Logger
-
 	config Config
-
 	listener net.Listener
 	server   http.Server
-
+	service  *marketing.Service
 	templateDir string
 }
 
-// commonPages returns templates that are required for everything.
+// Struct used to render each offer table
+type offerSet struct {
+	RefOffers,FreeCredits []marketing.Offer
+}
+
+// Takes db result of get offers, organizes them by type and returns them
+// Used by the index handler to build render the tables
+func organizeOffers(offers []marketing.Offer) offerSet{
+	os := new(offerSet)
```

Comment:
It's a little weird that we are initializing a pointer and then returning its value. Maybe we can do `var os offerSet` and then directly return its value at the end of this function

Reply:
You're right.

Reply:
done